rectools/models/candidate_ranking.py

```diff
@@ -0,0 +1,637 @@
+# flake8: noqa
+# TODO: docstrings
+
+import typing as tp
+from functools import reduce
+
+import attr
+import numpy as np
+import pandas as pd
+from catboost import CatBoostClassifier, CatBoostRanker, Pool
+
+from rectools import Columns
+from rectools.dataset import Dataset
+from rectools.dataset.identifiers import ExternalIds
+from rectools.model_selection import Splitter
+from rectools.models.base import ErrorBehaviour, ModelBase, NotFittedError
+
+
+@tp.runtime_checkable
+class ClassifierBase(tp.Protocol):
+    def fit(self, *args: tp.Any, **kwargs: tp.Any) -> None: ...
```

Comment:
1. Models usually return themselves after fit
2. Let's put x and y as explicit arguments, otherwise we can't check ourselves. Same for other methods

Reply:
1. Fixed
2. I don't see hot to put X and y in fit since they may have very diffent types. For example Catboost can accept Pool in X and have y as None.